code/game/objects/closets/walllocker.dm

```diff
@@ -1,58 +1,272 @@
+/obj/structure/closet/walllocker/medical_wall/full/west
+	pixel_y = 0
+	pixel_x = 32
+	dir = WEST
+
+/obj/structure/closet/walllocker/medical_wall/full/east
+	pixel_y = 0
+	pixel_x = -32
+	dir = EAST
+
+///////// SECURE WALL LOCKERS /////////
+
+/obj/structure/closet/secure_closet/walllocker
+	name = "secure wall locker"
+	desc = "It's an immobile card-locked storage unit."
+	icon = 'icons/obj/wallframes.dmi'
+	icon_state = "sec_locker1"
+	icon_closed = "secure"
+	icon_locked = "sec_locker1"
+	icon_opened = "sec_locker_opened"
+	icon_broken = "sec_locker_broken"
+	icon_off = "sec_locker_off"
+	overlay_welded = "emergwelded"
+	pixel_y = 32
+	density = FALSE
+	anchored = TRUE
+	store_mobs = FALSE
+	wall_mounted = TRUE
+	large = FALSE
+	storage_capacity = 20
+
+/obj/structure/closet/secure_closet/walllocker/north
+	pixel_y = -32
+	dir = NORTH
+
+/obj/structure/closet/secure_closet/walllocker/west
+	pixel_y = 0
 	pixel_x = 32
+	dir = WEST
+
+/obj/structure/closet/secure_closet/walllocker/east
+	pixel_y = 0
+	pixel_x = -32
+	dir = EAST
+
+/obj/structure/closet/secure_closet/walllocker/medical
+	name = "first aid closet"
+	desc = "It's a secure wall-mounted storage unit for first aid supplies."
+	icon_state = "medical_wall_sec1"
+	icon_closed = "medical_wall_sec"
+	icon_locked = "medical_wall_sec1"
+	icon_opened = "medical_wall_sec_open"
+	icon_broken = "medical_wall_sec_broken"
+	icon_off = "medical_wall_sec_off"
+	overlay_welded = "hydrant-medical_welded"
+	req_access = list(ACCESS_MARINE_MEDBAY)
+
+/obj/structure/closet/secure_closet/walllocker/medical/north
+	pixel_y = -32
+	dir = NORTH
+
+/obj/structure/closet/secure_closet/walllocker/medical/west
+	pixel_y = 0
+	pixel_x = 32
+	dir = WEST
+
+/obj/structure/closet/secure_closet/walllocker/medical/east
+	pixel_y = 0
+	pixel_x = -32
 	dir = EAST
+
+/obj/structure/closet/secure_closet/walllocker/medical/full // spawners
+
+/obj/structure/closet/secure_closet/walllocker/medical/full/New()
+	. = ..()
+	sleep(2)
+	new /obj/item/storage/syringe_case/tox(src)
+	new /obj/item/storage/syringe_case/oxy(src)
+	new /obj/item/storage/firstaid/adv(src)
+	new /obj/item/reagent_container/hypospray/autoinjector/quickclot(src)
+	new /obj/item/reagent_container/hypospray/autoinjector/dexalinplus(src)
+	new /obj/item/reagent_container/hypospray/autoinjector/oxycodone(src)
+
+/obj/structure/closet/secure_closet/personal/walllocker // different root path.
+	name = "personal wall locker"
+	desc = "It's a secure wall locker for personnel. The first card swiped gains control."
+	icon = 'icons/obj/wallframes.dmi'
+	icon_state = "sec_locker1"
+	icon_closed = "secure"
+	icon_locked = "sec_locker1"
+	icon_opened = "sec_locker_opened"
+	icon_broken = "sec_locker_broken"
+	icon_off = "sec_locker_off"
+	overlay_welded = "emergwelded"
+	pixel_y = 32
+	density = FALSE
+	anchored = TRUE
+	store_mobs = FALSE
+	wall_mounted = TRUE
+	large = FALSE
+	storage_capacity = 20
+
+/obj/structure/closet/secure_closet/personal/walllocker/north
```

Comment:
Don't do it like this (was discussed and deemed bad by spooky and alterist)

Reply:
Then, mind enlightening me a little on would this should be done?

Reply:
Direction should be varedited. Initialize should have a dir swith that sets the pixel offsets.

Reply:
okay dokay.